exonum-java-binding/qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -114,356 +88,394 @@ void tearDown() {
-    withNodeFake(() -> {
-      Throwable t = assertThrows(RuntimeException.class, () -> service.getBlockTransactions(0L));
-      assertThat(t.getMessage()).contains(NO_GENESIS_BLOCK_ERROR_MESSAGE);
-    });
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
+      List<HashCode> blockTransactions = service.getBlockTransactions(1L);
+      assertThat(blockTransactions).hasSize(1);
+      assertThat(blockTransactions.get(0)).isEqualTo(createCounterTransaction.hash());
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxMessages() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionMessage> txMessages = service.getTxMessages();
-      assertThat(txMessages).isEmpty();
-    });
+      assertThat(txMessages).hasSize(2);
+      assertThat(txMessages.get(createCounterTransaction.hash()))
+          .isEqualTo(createCounterTransaction);
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxResults() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionResult> txResults = service.getTxResults();
-      assertThat(txResults).isEmpty();
-    });
+      assertThat(txResults).hasSize(1);
+      assertThat(txResults.get(createCounterTransaction.hash()))
+          .isEqualTo(TransactionResult.successful());
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxResult() {
-    withNodeFake(() -> {
-      HashCode messageHash = HashCode.fromBytes(createPrefixed(bytes(0x00), 32));
-      Optional<TransactionResult> txResult = service.getTxResult(messageHash);
-      assertThat(txResult).isEmpty();
-    });
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
+      Optional<TransactionResult> txResult = service.getTxResult(createCounterTransaction.hash());
+      assertThat(txResult).contains(TransactionResult.successful());
```

Comment:
(here and below) I'd suggest using `hasValue` for Optionals to avoid confusion with collections and maps.